### PR TITLE
Bug 1212963 - Load -rtl variants of videos when dir=rtl. r=arcturus

### PR DESCRIPTION
**Pull request by sfoster**
https://bugzilla.mozilla.org/show_bug.cgi?id=1212963
- Refactor out helpers into a TutorialUtils library
- New helpers: fileExists, getBestAssetForDirection, loadAndPlayMedia
- Placeholder RTL video assets
- Unit tests for TutorialUtils library

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ffa83f7bcd9c92a60e6deb2423449d263ed14f36

File: apps/ftu/js/tutorial.js [hunk 1]
```diff
@@ -298,14 +236,15 @@
 
       var stepPromise;
       if (stepData.video) {
-        stepPromise = _loadMedia(videoElement, stepData.video).then(function(){
-          videoElement.play();
+        stepPromise = TutorialUtils.getBestAssetForDirection(stepData.video)
```
Comment (sfoster) [inline apps/ftu/js/tutorial.js]: This is the actual functional change - using getBestAssetForDirection to determine if we can get a direction-specific video (e.g. Homescreen-rtl.mp4 vs Homescreen.mp4) or just use what is in the config

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=845cd6d570782a75e0cc28f5de21382ba672df35

File: apps/ftu/js/tutorial_utils.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+  },
+
+  /**
+   * Helper to check (via xhr HEAD) a given resource exists
+   * @param {String} src  URL for video/image resource
+   * @returns {Promise}
+   */
+  fileExists: function (src) {
+    return new Promise((resolve, reject) => {
+      var xhr = new XMLHttpRequest();
```
Comment (arcturus) [inline apps/ftu/js/tutorial_utils.js]: Should we use `fetch` api that is already promise based?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c5dd4ed3a36201c6ade08e47532e30814c27d3fc

File: apps/ftu/js/tutorial_utils.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+  },
+
+  /**
+   * Helper to check (via HEAD request) a given resource exists
+   * The returned promise resolves with a true|false value
+   *
+   * @param {String} src  URL for video/image resource
+   * @returns {Promise}
+   */
+  fileExists: function (src) {
```
Comment (sfoster) [inline apps/ftu/js/tutorial_utils.js]: I switched out XHR for fetch here, which does end up much cleaner. But I want callers to be able to rely on a simple true|false result in their callback so I've wrapped fetch's promise here.

**Pull request merged**